File: api/src/main/java/br/cefet/vidralglass/model/Cliente.java
package br.cefet.vidralglass.model;

/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */

import java.util.List;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;
import lombok.experimental.SuperBuilder;

/**
 *
 * @author criad
 */
@Getter
@Setter
@ToString
@SuperBuilder
@NoArgsConstructor
@AllArgsConstructor
public class Cliente {
    protected int idCliente;
    protected String nome;
    protected String cpf;
    
    protected Endereco endereco;
}

File: src/app/model/cliente.ts
import { Endereco } from "./endereco";

export class Cliente {
    idCliente!: number;
    nome!: string;
    cpf!: string;
    endereco!: Endereco;

    constructor(){
        this.idCliente = 0;
        this.nome = "";
        this.cpf = "";
        this.endereco = new Endereco();
    }
}
